src/raster/r.fusion/r.fusion.py

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env python
+############################################################################
+#
+# MODULE:       r.fusion
+#
+# AUTHOR(S):    Markus Metz, mundialis
+#
+# PURPOSE:      Image fusion: similar to pan-sharpening, but more general
+#               burn the spatial detail of a high-res raster into a
+#               low-res raster, preserving the values of the low-res raster
+#
+# COPYRIGHT:    (c) 2023 Markus Metz and the GRASS development team
+#
+#               This program is free software under the GNU General Public
+#               License (>=v2). Read the file COPYING that comes with GRASS
+#               for details.
+#
+#############################################################################
+#
+
+# %module
+# % description: image fusion, generalized pan-sharpening
+# % keyword: raster
+# % keyword: imagery
+# % keyword: fusion
+# % keyword: sharpen
+# %end
+
+# %option G_OPT_R_INPUT
+# %  key: input
+# %  description: Low-resolution raster to be enhanced
+# %  required: yes
+# %end
+
+# %option G_OPT_R_INPUT
+# %  key: highres
+# %  description: High-resolution raster with spatial detail
+# %  required: yes
+# %end
+
+# %option
+# %  key: method
+# %  type: string
+# %  description: Solution method: Finite Diff. or Superpos. of analytical sol'ns
+# %  options: difference,proportion
+# %  answer: difference
+# %  required: no
+# %end
+
+# %option G_OPT_R_OUTPUT
+# %end
+
+# %flag
+# %  key: r
+# %  label: Restrict output value range to input value range
+# %  description: Recommended for imagery with a defined valid value range, e.g. [0, 255]
+# %end
+
+
+import atexit
+import os
+import sys
+
+import grass.script as gscript
+
+# initialize global vars
+rm_rasters = []
+current_region = None
+
+
+def cleanup():
+    nuldev = open(os.devnull, "w")
+    kwargs = {"flags": "f", "quiet": True, "stderr": nuldev}
+
+    # remove temporary raster maps
+    for rmrast in rm_rasters:
+        if gscript.find_file(name=rmrast, element="cell")["file"]:
+            gscript.run_command("g.remove", type="raster", name=rmrast, **kwargs)
+
+    # set region back to original region
+    if current_region:
+        gscript.run_command(
+            "g.region",
+            w=current_region["w"],
+            s=current_region["s"],
+            e=current_region["e"],
+            n=current_region["n"],
+            ewres=current_region["ewres"],
+            nsres=current_region["nsres"],
+            quiet=True,
+        )
+
+
+def main():
+    lowres_map = options["input"]
+    highres_map = options["highres"]
+    output_map = options["output"]
+    fusion_method = options["method"]
+    fit_range = flags["r"]
+
+    current_region = gscript.region()
```

Comment:
Why not use gscript.use_temp_region()?

Reply:
Because `gscript.use_temp_region()` sets an env var `WIND_OVERRIDE` which causes conflicts in nested scripts (script calling script calling script ...) using `use_temp_region()`, therefore we decided to not use `use_temp_region()` any more.

Reply:
As far as I remember, that is not true, see https://github.com/OSGeo/grass/pull/630#issuecomment-628870954

Reply:
Ah right, thanks for the reminder, changed accordingly.